modules/seqexec/web/server/src/main/scala/seqexec/web/server/http4s/WebServerLauncher.scala

```diff
@@ -187,33 +229,38 @@ object WebServerLauncher extends StreamApp[IO] with LogInitialization with Seqex
         met        <- SeqexecMetrics.build[IO](site, collector)
       } yield SeqexecEngine(httpClient, seqc, met)
 
-    def webServerIO(in: Queue[IO, executeEngine.EventType], out: Topic[IO, SeqexecEvent], et: SeqexecEngine, pe: PrometheusExportService[IO]): IO[Stream[IO, ExitCode]] =
-      // Launch web server
+    def webServerIO(
+      in:  Queue[IO, executeEngine.EventType],
+      out: Topic[IO, SeqexecEvent],
+      et:  SeqexecEngine,
+      pe:  PrometheusExportService[IO],
+      bec: ExecutionContext
+    ): Resource[IO, Unit] =
       for {
-        wc <- serverConf
-        ac <- authConf.run(wc)
-        as <- authService.run(ac)
-        _  <- logStart.run(wc)
-        _  <- logToClients(out)
-      } yield Stream(redirectWebServer.run(wc), webServer(as, in, out, et, pe).run(wc)).join(2)
-
-    // I have taken this from the examples at:
-    // https://github.com/gvolpe/advanced-http4s/blob/master/src/main/scala/com/github/gvolpe/fs2/PubSub.scala
-    // It's not very clear why we need to run this inside a Scheduler
-    Scheduler[IO](corePoolSize = 4).flatMap { implicit S =>
+        wc <- Resource.liftF(serverConf)
+        ac <- Resource.liftF(authConf)
+        as <- Resource.liftF(authService(ac))
+        _  <- Resource.liftF(logStart.run(wc))
+        _  <- Resource.liftF(logToClients(out))
+        _  <- redirectWebServer(wc)
+        _  <- webServer(as, in, out, et, pe, bec)(wc)
+      } yield ()
+
+    val r: Resource[IO, ExitCode] =
       for {
-        cli    <- Http1Client.stream[IO]()
-        inq    <- Stream.eval(async.boundedQueue[IO, executeEngine.EventType](10))
-        out    <- Stream.eval(async.topic[IO, SeqexecEvent](NullEvent))
-        pe     <- Stream.eval(PrometheusExportService.build[IO])
-        engine <- Stream.eval(engineIO(cli, pe.collectorRegistry))
-        web    <- Stream.eval(webServerIO(inq, out, engine, pe))
-        exit   <- Stream(
-                    engine.eventStream(inq).to(out.publish),
-                    web
-                  ).join(2).drain ++ Stream.emit(ExitCode.Success)
-      } yield exit
-    }
+        cli    <- BlazeClientBuilder[IO](null).resource
```

Comment:
`null`?

Reply:
Thanks, I needed to pass an EC there and stubbed it out. I think that explains another issue I'm seeing.